### PR TITLE
Lz4 recipe

### PR DESCRIPTION
**Pull request by esben**

### CODE REVIEW

File: recipes/lz4/lz4_r131.oe [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require ${PN}.inc
+
+SRC_URI += "file://windows-h-include.patch"
+SRC_URI += "file://exeext.patch"
```
Comment (kimrhh) [inline recipes/lz4/lz4_r131.oe]: shouldn't they all be conditional on mingw?

File: recipes/lz4/lz4.inc [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+S = "${SRCDIR}/lz4-${PV}"
+
+inherit c make
+
+#EXTRA_OEMAKE = "PREFIX=${prefix} CC='${CC}' DESTDIR=${D} LIBDIR=${libdir} INCLUDEDIR=${includedir}"
+#EXTRA_OEMAKE = "PREFIX=${prefix} CC='${CC}' DESTDIR=${D} LIBDIR=${libdir} INCLUDEDIR=${includedir}"
+#
+#do_install() {
+#	oe_runmake install
+#}
```
Comment (kimrhh) [inline recipes/lz4/lz4.inc]: remove commented dev code? :)

File: recipes/lz4/lz4_r131.oe (hunk 1 above)
Comment (esben) [inline recipes/lz4/lz4_r131.oe]: Good question.

Pro: They are only relevant for mingw.

Con: Developers not doing mingw build themselves will see patch conflicts if we apply mingw (and similar for osx patches) for linux builds also.

Let me know what you want here.

File: recipes/lz4/lz4.inc (hunk 2 above)
Comment (esben) [inline recipes/lz4/lz4.inc]: Definitely.

File: recipes/lz4/lz4_r131.oe (hunk 1 above)
Comment (kimrhh) [inline recipes/lz4/lz4_r131.oe]: I don't understand your con reasoning :) if you don't build for mingw, and those are applied conditionally, how would that couse a patch conflict? If I added another patch only for Linux builds?

Comment (esben) [inline recipes/lz4/lz4_r131.oe]: Double (or more) negation sucks :)

Pro applying mingw patches conditionally: They are only relevant for mingw.

Pro applying mingw patches unconditionally: Developers not doing mingw build themselves will see patch conflicts for linux builds also.  Patch is closer to a form that can be accepted by upstream.

**Comment by esben** [comment]
Ready for merge?

**Pull request merged**